Wikipedia/Code/SearchViewController.swift

```diff
@@ -97,7 +97,9 @@ class SearchViewController: ArticleCollectionViewController, UISearchBarDelegate
         let start = Date()
         
         AFNetworkActivityIndicatorManager.shared().incrementActivityCount()
-        fakeProgressController.start()
+        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(1000)) {
```

Comment:
If the user types a bunch, this will still call `start()` for every key. I thought the intent was to call `start()` 1s after the user has been idle. To do this, it might be best to integrate this into `FakeProgressController` as `cancelPreviousStartsAndStartAfter(_ after: TimeInterval)` and use `perform(` and `cancelPreviousPerformRequests(`